wrapanapi/systems/virtualcenter.py

```diff
@@ -241,14 +286,166 @@ def _get_resource_pool(self, resource_pool_name=None):
             return self.system.get_obj(vim.ResourcePool, resource_pool_name)
         elif self.system.default_resource_pool is not None:
             return self.system.get_obj(vim.ResourcePool, self.system.default_resource_pool)
-        return self.system.get_obj_list(vim.ResourcePool)[0]
+        else:
+            return self.system.get_obj_list(vim.ResourcePool)[0]
+
+    def _get_cluster_compute_resource(self, cluster_compute_resource_name=None):
+        """ Returns a Compute Cluster Resource managed object. If a name is specified,
+        a vim.ClusterComputeResource object is returned for the specific resource. If no name is
+        specified, the method checks if there are is a default resource specified and returns the
+        object of the resource. Finally,  if there is no name or defaults specified, it queries
+        the Vmware provider and picks the first object returned in the list.
+        Args:
+            cluster_compute_resource_name (string): The name of the Cluster Compute Resource.
+            If None, first one will be picked.
+        Returns:
+             pyVmomi.vim.ClusterComputeResource: The managed object of the cluster compute resource.
+        """
+        if cluster_compute_resource_name is not None:
```

Comment:
you could optimize all this block

Reply:
Leaving it as is after discussion offline